source/client-side-encryption/tests/README.md

```diff
@@ -3296,3 +3296,65 @@ Assert the returned payload size is greater than the size of `payload_defaults`.
 > [!NOTE]
 > Do not compare the payload contents. The payloads include random data. The `trimFactor` and `sparsity` directly affect
 > the payload size.
+
+### 24. KMS Retry Tests
+
+The following tests that certain AWS, Azure, and GCP KMS operations are retried on transient errors.
+
+This test uses a mock server with configurable failpoints to simulate network failures. To start the server:
+
+```
+python -u kms_failpoint_server.py --port 9003
+```
+
+See the [TLS tests](#10-kms-tls-tests) for running the mock server on Evergreen. See
+[the mock server implementation](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/4ba50d373652b6fb39239745664637e33e2b01e6/.evergreen/csfle/kms_failpoint_server.py)
+and the [C driver tests](TODO) for how to configure failpoints.
+
+#### Setup
+
+1. Start a `mongod` process with **server version 4.2.0 or later**.
+2. Create a `MongoClient` for key vault operations.
+3. Create a `ClientEncryption` object (referred to as `client_encryption`) with `keyVaultNamespace` set to
+   `keyvault.datakeys`.
+
+#### createDataKey
```

Comment:
Suggest separating HTTP and TCP test cases to isolate tested behavior (drivers are required to call `mongocrypt_kms_ctx_fail` for TCP retry, so there may be value in separate testing).

Suggest also adding a test case that requires too-many retry attempts. That may help verify the test set-up is correctly configuring the mock server:

```md
#### Case 1: createDataKey with TCP retry
...
#### Case 2: createDataKey with TCP retry
...
#### Case 3: createDataKey fails after too many retries
...
```